service/dps/storm/common/src/main/java/eu/europeana/cloud/service/dps/storm/spouts/kafka/job/TaskExecutor.java

```diff
@@ -0,0 +1,335 @@
+import org.slf4j.LoggerFactory;
+
+import java.net.MalformedURLException;
+import java.util.ConcurrentModificationException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.*;
+
+import static eu.europeana.cloud.service.dps.storm.AbstractDpsBolt.NOTIFICATION_STREAM_NAME;
+
+public class TaskExecutor implements Callable<Void> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(TaskExecutor.class);
+    private static final int INTERNAL_THREADS_NUMBER = 10;
+    private static final int DEFAULT_RETRIES = 3;
+    private static final int SLEEP_TIME = 5000;
+    public static final int MAX_BATCH_SIZE = 100;
+
+    private TaskStatusChecker taskStatusChecker;
+
+    private SpoutOutputCollector collector;
+    private CassandraTaskInfoDAO cassandraTaskInfoDAO;
+    private ArrayBlockingQueue<StormTaskTuple> tuplesWithFileUrls;
+
+    private DataSetServiceClient dataSetServiceClient;
+    private RecordServiceClient recordServiceClient;
+    private FileServiceClient fileClient;
+
+    private String mcsClientURL;
+
+    private String stream;
+    private DpsTask dpsTask;
+
+    public TaskExecutor(SpoutOutputCollector collector, TaskStatusChecker taskStatusChecker, CassandraTaskInfoDAO cassandraTaskInfoDAO, ArrayBlockingQueue<StormTaskTuple> tuplesWithFileUrls, DataSetServiceClient dataSetServiceClient, RecordServiceClient recordServiceClient, FileServiceClient fileClient, String mcsClientURL, String stream, DpsTask dpsTask) {
+        this.collector = collector;
+        this.taskStatusChecker = taskStatusChecker;
+        this.cassandraTaskInfoDAO = cassandraTaskInfoDAO;
+        this.tuplesWithFileUrls = tuplesWithFileUrls;
+
+        this.dataSetServiceClient = dataSetServiceClient;
+        this.recordServiceClient = recordServiceClient;
+        this.fileClient = fileClient;
+
+        this.mcsClientURL = mcsClientURL;
+
+        this.stream = stream;
+        this.dpsTask = dpsTask;
+    }
+
+    @Override
+    public Void call() throws Exception {
+        execute();
+        return null;
+    }
+
+    private void execute() throws Exception {
+        final List<String> dataSets = dpsTask.getDataEntry(InputDataType.valueOf(stream));
+        final String representationName = dpsTask.getParameter(PluginParameterKeys.REPRESENTATION_NAME);
+        dpsTask.getParameters().remove(PluginParameterKeys.REPRESENTATION_NAME);
+
+        final String revisionName = dpsTask.getParameter(PluginParameterKeys.REVISION_NAME);
+        dpsTask.getParameters().remove(PluginParameterKeys.REVISION_NAME);
+
+        final String revisionProvider = dpsTask.getParameter(PluginParameterKeys.REVISION_PROVIDER);
+        dpsTask.getParameters().remove(PluginParameterKeys.REVISION_PROVIDER);
+
+        final String authorizationHeader = dpsTask.getParameter(PluginParameterKeys.AUTHORIZATION_HEADER);
+        dataSetServiceClient.useAuthorizationHeader(authorizationHeader);
```

Comment:
we only have one instance of DataSetServiceClient, RecordServiceClient and FileServiceClient, so we can't change the authorizationHeader per thread. this will affect the other threads.